ecosystem/indexer-server/typescript/src/utils/openapiRouter.ts

```diff
@@ -0,0 +1,77 @@
+import logger from '../logger';
+import controllers from '../controllers';
+import Services from '../services';
+
+function handleError(
+  err: any,
+  request: any,
+  response: any,
+  next: (props: any) => void,
+) {
+  logger.error(err);
+  const code = err.code || 400;
+  response.status(code);
+  response.error = err;
+  next(JSON.stringify({
+    code,
+    error: err,
+  }));
+}
+
+/**
+ * The purpose of this route is to collect the request variables as defined in the
+ * OpenAPI document and pass them to the handling controller as another Express
+ * middleware. All parameters are collected in the request.swagger.values key-value object
+ *
+ * The assumption is that security handlers have already verified and allowed access
+ * to this path. If the business-logic of a particular path is dependant on authentication
+ * parameters (e.g. scope checking) - it is recommended to define the authentication header
+ * as one of the parameters expected in the OpenAPI/Swagger document.
+ *
+ *  Requests made to paths that are not in the OpenAPI scope
+ *  are passed on to the next middleware handler.
+ * @returns {Function}
+ */
+function openApiRouter() {
+  return async (request: any, response: any, next: () => void) => {
+    console.log('hi');
+    try {
+      /**
+       * This middleware runs after a previous process have applied an openapi object
+       * to the request.
+       * If none was applied This is because the path requested is not in the schema.
+       * If there's no openapi object, we have nothing to do, and pass on to next middleware.
+       */
+      if (request.openapi === undefined
+          || request.openapi.schema === undefined
+      ) {
+        next();
+        return;
```

Comment:
this feels weird, but I don't know what next does

Reply:
getting rid of this file